hal/src/stm32f2xx/spi_hal.c

```diff
@@ -147,7 +159,7 @@ static void HAL_SPI_DMA_Config(HAL_SPI_Interface spi, void* tx_buffer, void* rx_
     }
     else
     {
-        DMA_InitStructure.DMA_Memory0BaseAddr = (uint32_t)(&tempMemory);
+        DMA_InitStructure.DMA_Memory0BaseAddr = (uint32_t)(&tempMemoryTx);
```

Comment:
omg! Nice catch. How did this ever work before? using a stack variable as the DMA address!